### PR TITLE
refactor: Reuse the the builder configuration in local mode

### PR DESCRIPTION
**Pull request by HofmeisterAn**
Hi @nkz-soft 👋 I finally had some time to look into and test the pull request related to the Compose feature. While reviewing and testing the PR, I made some changes that I would like to discuss with you first:

1. The [DockerComposeBuilder.Build()](https://github.com/testcontainers/testcontainers-dotnet/blob/90f6ccbada5e6d92b92848fddab7fd85c272b047/src/Testcontainers.DockerCompose/DockerComposeBuilder.cs#L65-L68) member can determine which `DockerComposeContainer` implementation should be returned. I don't believe we need `DockerCompose` and the proxy implementation. Additionally, to 2. this reduces the amount of necessary configurations.
2. I propose to [reuse](https://github.com/testcontainers/testcontainers-dotnet/blob/90f6ccbada5e6d92b92848fddab7fd85c272b047/src/Testcontainers.DockerCompose/DockerComposeLocalContainer.cs#L24-L28) the builder configuration in the local Docker Compose mode. The command line builder is probably unnecessary as well. The idea is to set the [entrypoint configuration](https://github.com/testcontainers/testcontainers-dotnet/blob/90f6ccbada5e6d92b92848fddab7fd85c272b047/src/Testcontainers.DockerCompose/DockerComposeBuilder.cs#L41-L46) to the compose command, such as `docker compose --file docker-compose.yml`, and use the [command configuration](https://github.com/testcontainers/testcontainers-dotnet/blob/90f6ccbada5e6d92b92848fddab7fd85c272b047/src/Testcontainers.DockerCompose/DockerComposeBuilder.cs#L79-L80) for the actual compose command options, e.g., `up [OPTIONS]`. This has the advantage that developers can easily append options using the `WithCommand(string)` API.
    - Using this approach makes exposing additional (compose) options, such as `RemoveImages`, unnecessary as well. We can implement `WithRemoveImages(RemoveImages)` simply by:
    ```csharp
    public DockerComposeBuilder WithCleanUp(RemoveImages removeImages)
    {
        return WithCommand("--rmi", removeImages);
    }
    ```

WDYT? Out of simplicity, I removed some parts related to the tests. I would like to discuss a few things first before we finalize the tests again.

### CODE REVIEW

File: src/Testcontainers.DockerCompose/DockerComposeBuilder.cs [hunk 1]
```diff
@@ -2,18 +2,17 @@ namespace Testcontainers.DockerCompose;
...
+public sealed class DockerComposeBuilder : ContainerBuilder<DockerComposeBuilder, DockerComposeContainer, DockerComposeConfiguration>
 {
-    private const string NoComposeFile = "No docker compose file have been provided.";
+    public const string DockerComposeFilePath = "/docker-compose.yml";
 
-    //Docker Compose is included as part of this image.
-    private const string DockerComposeImage = "docker:24-cli";
+    public const string DockerComposeImage = "docker:25.0-cli";
+
+    // TODO: This does not support all container runtimes (host configurations). We should do something similar to what we are doing in the Resource Reaper implementation.
```
Comment (HofmeisterAn) [inline src/Testcontainers.DockerCompose/DockerComposeBuilder.cs:11]: I can take care of this.

File: src/Testcontainers.DockerCompose/DockerComposeBuilder.cs [hunk 2]
```diff
@@ -34,68 +33,52 @@ private DockerComposeBuilder(DockerComposeConfiguration resourceConfiguration)
...
     {
         return base.Init()
             .WithImage(DockerComposeImage)
-            .WithEntrypoint(DockerComposeCommandLineBuilder.DockerAppName)
-            .WithCommand(DockerComposeCommandLineBuilder
-                .FromRemoteConfiguration(DockerResourceConfiguration)
-                .BuildStartCommand()
-                .ToArray())
+            .WithCommand("up")
+            .WithCommand("--detach")
```
Comment (HofmeisterAn) [inline src/Testcontainers.DockerCompose/DockerComposeBuilder.cs:80]: We need to move this to local mode only.

File: src/Testcontainers.DockerCompose/DockerComposeRemoteContainer.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DockerComposeRemoteContainer" /> class.
+    /// </summary>
+    /// <param name="configuration">The container configuration.</param>
+    /// <param name="logger">The logger.</param>
+    public DockerComposeRemoteContainer(DockerComposeConfiguration configuration, ILogger logger)
+        : base(configuration, logger)
+    {
+    }
+}
```
Comment (HofmeisterAn) [inline src/Testcontainers.DockerCompose/DockerComposeRemoteContainer.cs:15]: Override `StopAsync(CancellationToken)` again.

File: tests/Testcontainers.DockerCompose.Tests/DockerComposeTest.cs [hunk 4]
```diff
@@ -1,102 +1,54 @@
...
+        return Task.CompletedTask;
+        return _dockerComposeContainer.DisposeAsync().AsTask();
     }
-    
+
     [Fact]
     [Trait(nameof(DockerCli.DockerPlatform), nameof(DockerCli.DockerPlatform.Linux))]
     public void ContainerStartedSuccessfully()
     {
+        // TODO: How do we test that the Compose configuration is actually working? How do we access services?
```
Comment (HofmeisterAn) [inline tests/Testcontainers.DockerCompose.Tests/DockerComposeTest.cs:27]: How can we actually test if the configuration is running? How do we obtain the endpoint of a service, such as the actual hostname and port?

File: src/Testcontainers.DockerCompose/DockerComposeBuilder.cs [hunk 5]
```diff
@@ -34,68 +33,52 @@ private DockerComposeBuilder(DockerComposeConfiguration resourceConfiguration)
...
     protected override DockerComposeBuilder Init()
     {
         return base.Init()
             .WithImage(DockerComposeImage)
-            .WithEntrypoint(DockerComposeCommandLineBuilder.DockerAppName)
-            .WithCommand(DockerComposeCommandLineBuilder
-                .FromRemoteConfiguration(DockerResourceConfiguration)
-                .BuildStartCommand()
-                .ToArray())
+            .WithCommand("up")
```
Comment (HofmeisterAn) [inline src/Testcontainers.DockerCompose/DockerComposeBuilder.cs:79]: How do we know the container is up and running ([ready](https://dotnet.testcontainers.org/api/wait_strategies/))? How do we know that the services from the Compose file are ready and accepting connections, etc.?

File: tests/Testcontainers.DockerCompose.Tests/DockerComposeTest.cs (hunk 4 above)
Comment (nkz-soft) [reply tests/Testcontainers.DockerCompose.Tests/DockerComposeTest.cs:27]: To fully support tests with services or containers, we need to introduce an additional abstraction, like [listChildContainers ](https://github.com/testcontainers/testcontainers-java/blob/5969b0f393b724c9ed2147272d87a24c8d9efe1c/core/src/main/java/org/testcontainers/containers/ComposeDelegate.java#L264) in Java. At the moment I would limit myself to checking availability using healthchecks in the file itself.

**Comment by nkz-soft** [comment]
Hi @HofmeisterAn.

Unfortunately, I've been very busy for the last month. 
Let's agree on which Java implementation features we will support?

The `DockerComposeCommandLineBuilder` class was made just to be able to handle methods like `WithOptions`, `RemoveImages `and others. The main idea was that the options can be applied both at startup (which can be replaced by calling `WithCommand`) and when the container is stopped (e.g. -rmi). If we don't support these features in this version, there's no question. But I think we will have to come back to it later anyway.

Anyway, this seems like a first step and we can add the necessary functionality later if needed.

**Pull request merged**